### PR TITLE
New transcript feed format

### PR DESCRIPTION
**Pull request by jaril**
image

### CODE REVIEW

File: src/ui/actions/timeline.ts [hunk 1]
```diff
@@ -176,6 +176,7 @@ export function seek(
     if (pause) {
       ThreadFront.timeWarpToPause(pause);
     } else {
+      console.log("warping to", point, time);
```
Comment (jasonLaster) [inline src/ui/actions/timeline.ts]: ^

File: src/ui/components/Transcript/AddCommentButton.tsx [hunk 2]
```diff
@@ -27,8 +27,10 @@ function AddCommentButton({
       return setModal("login");
     }
 
+    console.log(">>point", ThreadFront.currentPoint);
```
Comment (jasonLaster) [inline src/ui/components/Transcript/AddCommentButton.tsx]: ^

File: src/ui/reducers/app.ts [hunk 3]
```diff
@@ -6,9 +6,12 @@ const { prefs } = require("../utils/prefs");
 import { Location } from "@recordreplay/protocol";
 import { getLocationKey } from "devtools/client/debugger/src/utils/breakpoint";
 
+const url = new URL(window.location.href);
+const recordingId = url.searchParams.get("id");
+
 function initialAppState(): AppState {
   return {
-    recordingId: null,
+    recordingId: recordingId!,
```
Comment (jasonLaster) [inline src/ui/reducers/app.ts:14]: This is fine for now, but it'd be better to setup app's initial state properly.

This ball of code will move `initialAppState` to `bootstrapStore`...

```diff
diff --git a/src/main.js b/src/main.js
index 62489ca5..c5468942 100644
--- a/src/main.js
+++ b/src/main.js
@@ -70,7 +70,7 @@ async function initialize() {

 (async () => {
   window.gToolbox = new DevToolsToolbox();
-  store = await bootstrapStore();
+  store = await bootstrapStore({ recordingId });

   if (!initialized) {
     initialized = true;
diff --git a/src/ui/reducers/app.ts b/src/ui/reducers/app.ts
index 9f653710..1c688ecb 100644
--- a/src/ui/reducers/app.ts
+++ b/src/ui/reducers/app.ts
@@ -34,10 +34,7 @@ function initialAppState(): AppState {
   };
 }

-export default function update(
-  state = initialAppState(),
-  action: AppActions | SessionActions
-): AppState {
+export default function update(state: AppState, action: AppActions | SessionActions): AppState {
   switch (action.type) {
     case "setup_app": {
       return { ...state, recordingId: action.recordingId };
diff --git a/src/ui/utils/bootstrap/bootstrapStore.js b/src/ui/utils/bootstrap/bootstrapStore.js
index 30b991f7..d3e4c237 100644
--- a/src/ui/utils/bootstrap/bootstrapStore.js
+++ b/src/ui/utils/bootstrap/bootstrapStore.js
@@ -15,12 +15,15 @@ import { sanityCheckMiddleware } from "../sanitize";
 import { isDevelopment, isTest } from "../environment";
 const skipTelemetry = isTest() || isDevelopment();

-async function getInitialState() {
+async function getInitialState({ recordingId }) {
   const eventListenerBreakpoints = await asyncStore.eventListenerBreakpoints;
   const initialDebuggerState = await dbgClient.loadInitialState();
   const initialConsoleState = getConsoleInitialState();

+  const app = initialAppState({ recordingId });
+
   return {
+    app,
     ...initialDebuggerState,
     eventListenerBreakpoints,
     ...initialConsoleState,
@@ -54,7 +57,7 @@ function updatePrefs(state, oldState) {
   updateAsyncPref("eventListenerBreakpoints", state => state.eventListenerBreakpoints);
 }

-export const bootstrapStore = async function bootstrapStore() {
+export const bootstrapStore = async function bootstrapStore({ recordingId }) {
   const debuggerWorkers = bootstrapWorkers();

   // TODO; manage panels outside of the Toolbox componenet
@@ -75,7 +78,7 @@ export const bootstrapStore = async function bootstrapStore() {
     },
   });

-  const initialState = await getInitialState();
+  const initialState = await getInitialState({ recordingId });
   const middleware = skipTelemetry
     ? isDevelopment()
       ? applyMiddleware(sanityCheckMiddleware)
```

**Review by jasonLaster — APPROVED** [review]
Looks great!

File: src/ui/reducers/app.ts (hunk 3 above)
Comment (jaril) [reply src/ui/reducers/app.ts]: Found a different workaround for the type error so we didn't have to do it this way. 

Redux isn't accommodating of specifying initial state for one reducer out of a couple that we combine with `combineReducer`, so this gets hairy.

**Review by jasonLaster — APPROVED** [review]

**Pull request merged**